controllers/discordactions.js

```diff
@@ -0,0 +1,128 @@
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const admin = require("firebase-admin");
+const config = require("config");
+const jwt = require("jsonwebtoken");
+const discordRolesModel = require("../models/discordactions");
+
+/**
+ * Creates a role
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+
+const createGroupRole = async (req, res) => {
+  try {
+    const rolename = `group-${req.body.rolename}`;
+
+    const { wasSuccess } = await discordRolesModel.isGroupRoleExists(rolename);
+
+    if (!wasSuccess) {
+      return res.status(400).json({
+        message: "Role already exists!",
+      });
+    }
+    const dataForDiscord = {
+      rolename,
+      mentionable: true,
+    };
+    const groupRoleData = {
+      rolename,
+      createdBy: req.userData.id,
+      date: admin.firestore.Timestamp.fromDate(new Date()),
+    };
+    const authToken = jwt.sign({}, config.get("rdsServerlessBot.rdsServerLessPrivateKey"), {
+      algorithm: "RS256",
+      expiresIn: config.get("userToken.ttl"),
```

Comment:
let's keep this for a smaller duration, please. userToken exists for a longer duration.

Our cloudflare bot will only be active for 10 ms so I think the max age of token can be 1 minute